### DISCUSSION TITLE
The right socket.io version cloent to use

### DISCUSSION THREAD

**Discussion by Notadumdum** [miguelgrinberg/python-socketio#1055] [Q&A]
Hey Miguel I heard python doesnt work well with browser based client and that i must use javascript
I just have one question.. please what version of socket.io should i use and must it tally with the exact version the server uses... note that the URL has EIO=3 in it...and when i ran code multiple times and checked the reason it showed ```transport close```... i did not even get any sid back to begin with.. the disconnect event was what was always called.. any advice?

**Comment by bhakta0007** [comment]
please follow this version guide and it will just work fine:

https://python-socketio.readthedocs.io/en/latest/intro.html#version-compatibility


this is what I am using:

JS:
```
grep -i socket package.json
    "socket.io-client": "^4.5.1",
```

Python:
```
➜ grep -i socket requirements.txt
Flask-SocketIO==5.1.2
python-socketio==5.6.0
socketIO-client==0.7.2
websocket-client==0.57.0
```

**Reply by Notadumdum** [reply]
@bhakta0007 i am using javascript now and the url contains an ```EIO=3```
So what version would i use and also could i get ypur discord name ive got multiple questions

**Reply by bhakta0007** [reply]
EIO=3 is engine io version 3 [https://github.com/socketio/engine.io-protocol/tree/v3]

So lookup row 2 of that Version compatibility and ensure your packages match.

sorry, I don't have discord. If you can post your questions here, I can answer them

**Reply by Notadumdum** [reply]
@bhakta0007 
![20221004_152201](https://user-images.githubusercontent.com/105162586/193845186-cf0caff7-363f-44ee-a05d-22783d70b1fd.jpg)

I'm trying to connect to a server using these messages being sent and sadly i have not really been able to..
But the socket url looks like this

```wss://ws2.qxbroker.com/socket.io/?EIO=3&transport=websocket```

![Screenshot_20221004-154209_Kiwi Browser](https://user-images.githubusercontent.com/105162586/193850570-03196f57-1fb0-45ad-ae23-d1ef8ca9e911.jpg)
These are the kinda stuff received by the browser..
 You could check it out for yourself
Qxbroker.com

**Reply by Notadumdum** [reply]
```js
const socket = require('socket.io-client')('https://ws2.qxbroker.com/socket.io/')

socket.on('connect',() => {
    console.log('connected to the server')

})

socket.on('event',(data) => {
    socket.emit("authorization",JSON.stringify({"session":"rrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrtrrr","isDemo":1}))
    socket.emit("tick")
    socket.emit("indicator/list")
    socket.emit("drawing/load")
    socket.emit("pending/list")
    socket.emit("instruments/update",JSON.stringify({"asset":"NZDUSD","period":60}))
    socket.emit("chart_notification/get")
    socket.emit("depth/follow","NZDUSD")
    socket.emit("settings/store",JSON.stringify({"chartId":"graph","settings":{"chartId":"graph","chartType":2,"currentExpirationTime":0,"isFastOption":false,"isFastAmountOption":false,"isIndicatorsMinimized":false,"isIndicatorsShowing":true,"isShortBetElement":false,"chartPeriod":4,"currentAsset":{"symbol":"NZDUSD"},"dealValue":1000,"dealPercentValue":1,"isVisible":true,"timePeriod":60,"gridOpacity":8,"isAutoScrolling":1,"isOneClickTrade":true,"upColor":"#0FAF59","downColor":"#FF6251"}}))
    console.log('event:' + data)
})

socket.on('disconnect',() => {
    console.log('disconnected from server')
})
```

@bhakta0007  this is the code i used earlier

**Reply by bhakta0007** [reply]
What kind of  socket io server is the qxbroker running? When i tried a sample program to connect to the server, I See 

```
[2022-10-05 13:40:48] DEBUG - asyncio::__init__ - Using selector: EpollSelector
[2022-10-05 13:40:48] INFO - root::start_client - Try to connect to wss://ws2.qxbroker.com/socket.io/
[2022-10-05 13:40:48] INFO - engineio.client::_connect_polling - Attempting polling connection to https://ws2.qxbroker.com/socket.io/?transport=polling&EIO=3
[2022-10-05 13:40:48] INFO - engineio.client::_connect_polling - Polling connection accepted with {'sid': 'SQQYJqHuUfcRpmFWBY75', 'upgrades': ['websocket'], 'pingInterval': 25000, 'pingTimeout': 5000}
[2022-10-05 13:40:48] INFO - socketio.client::_handle_eio_connect - Engine.IO connection established
[2022-10-05 13:40:48] INFO - engineio.client::_receive_packet - Received packet MESSAGE data 0
[2022-10-05 13:40:48] INFO - engineio.client::_connect_websocket - Attempting WebSocket upgrade to wss://ws2.qxbroker.com/socket.io/?transport=websocket&EIO=3
[2022-10-05 13:40:48] INFO - socketio.client::_handle_connect - Namespace / is connected
[2022-10-05 13:40:48] INFO - root::connect - connected to server
[2022-10-05 13:40:48] INFO - root::connect - Started BG thread
[2022-10-05 13:40:48] INFO - root::background_thread - background_thread
[2022-10-05 13:40:49] WARNING - engineio.client::_connect_websocket - WebSocket upgrade failed: connection error
```


how does one authenticate with the server? I see from your code above that you are sending authorization on an event. How is this supposed to work? Can you elaborate a bit on what the server is and how it expects users to connect and authorize?

I would have though that you would be sending some headers containing authorization during connect itself.

**Reply by Notadumdum** [reply]
Its amazing that you could get an sid{} sent back to you.. this code went straight to disconnect for me...i dont really know how the server is meant to authenticate.. i'm basically trying to replicate the connection using what i could see in "qxbroker.com" devtools..also please did you change anything in the code and what version of socket.io is this @bhakta0007

**Reply by Notadumdum** [reply]
![20221005_193030](https://user-images.githubusercontent.com/105162586/194135706-8dda077a-5410-4b1a-b6f1-4467bbbb7fc6.jpg)

Also would this stylesheet help in any way @bhakta0007 
Please can I have your email

**Reply by Notadumdum** [reply]
If it helps in anyway the server used is nginx @bhakta0007

**Comment by bhakta0007** [comment]


**Reply by Notadumdum** [reply]
I am solicit#3494
